### ISSUE TITLE
Problem in handling messages from PubSub concurrently

### ISSUE THREAD

**Issue by mantishK** [GoogleCloudPlatform/golang-samples#1455]
Referring to the code below 
https://github.com/GoogleCloudPlatform/golang-samples/blob/a289aabbb0bb9c8f720e22f61b252874970e57f4/pubsub/subscriptions/pull_concurrency.go#L28

I recently came across a case where the receiver blocked indefinitely. I used the same example with a context timeout of 10s. What happened was, the `ctx.Done()` on line 58 was getting executed before the receiver stopped. So, any messages that came in later (and before receiver stopped) were not getting acknowledged and hence the receiver blocked.

Is it really necessary to handle `ctx.Done()` in the function again? Can't we write the anonymous function in the following way -

```
go func() {
    for {
        msg, ok: = < -cm
        if !ok {
            return
        }
        fmt.Fprintf(w, "Got message :%q\n", string(msg.Data))
        msg.Ack()
    }
}()
```

I am new to concurrency in Go, maybe I am missing something.

**Comment by hongalex** [comment]
Yeah, I think your proposal makes sense and is a bit cleaner. If you want to create a PR to fix this, I'd be happy to review it. Otherwise, I'll get to this when I have some free time (since there are a few other samples that could use updating too).

**Comment by mantishK** [comment]
Raised a PR [here](https://github.com/GoogleCloudPlatform/golang-samples/pull/1467)
